services/src/main/java/org/keycloak/storage/ClientStorageManager.java

```diff
@@ -148,16 +159,22 @@ public ClientModel getClientByClientId(RealmModel realm, String clientId) {
     }
 
     @Override
-    public List<ClientModel> searchClientsByClientId(RealmModel realm, String clientId, Integer firstResult, Integer maxResults) {
-        List<ClientModel> clients = session.clientLocalStorage().searchClientsByClientId(realm, clientId,  firstResult, maxResults);
-        if (clients != null) {
-            return clients;
-        }
-        for (ClientLookupProvider provider : getEnabledStorageProviders(session, realm, ClientLookupProvider.class)) {
-            clients = provider.searchClientsByClientId(realm, clientId, firstResult, maxResults);
-            if (clients != null) return clients;
+    public Stream<ClientModel> searchClientsByClientIdStream(RealmModel realm, String clientId, Integer firstResult, Integer maxResults) {
```

Comment:
This method IMO should have also javadoc which describes its behavior, because on the first look it's not so obvious.

Reply:
done